app/controllers/BaseController.java

```diff
@@ -49,19 +49,15 @@
 
     private static ObjectMapper mapper = BridgeObjectMapper.get();
 
-    protected AuthenticationService authenticationService;
-    protected StudyService studyService;
-    protected CacheProvider cacheProvider;
-    protected BridgeConfig bridgeConfig;
+    private BridgeConfig bridgeConfig;
+    private CacheProvider cacheProvider;
 
-    @Autowired
-    public void setStudyService(StudyService studyService) {
-        this.studyService = studyService;
-    }
+    StudyService studyService;
```

Comment:
You blew my mind with this one. I looked online and you're right: https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html The only difference is visibility by subclasses. Crazy.